src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.MvcApplicationPartsDiscovery.targets

```diff
@@ -0,0 +1,103 @@
+
+    <CoreCompileDependsOn Condition="'$(GenerateMvcApplicationPartsAttribute)' == 'true' AND '$(DesignTimeBuild)' != 'true'">
+      _DiscoverMvcApplicationParts;
+      $(CoreCompileDependsOn);
+    </CoreCompileDependsOn>
+
+    <_MvcApplicationPartAttributeGeneratedFile>$(IntermediateOutputPath)$(TargetName).MvcApplicationPartsAssemblyInfo$(DefaultLanguageSourceExtension)</_MvcApplicationPartAttributeGeneratedFile>
+
+    <_DiscoverMvcApplicationPartsDependsOn>
+      ResolveAssemblyReferences;
+      _InitializeDiscoverMvcApplicationParts;
+    </_DiscoverMvcApplicationPartsDependsOn>
+  </PropertyGroup>
+
+  <Target Name="_InitializeDiscoverMvcApplicationParts">
+    <!-- Can be used by testassets which need to simulate a reference to Microsoft.AspNetCore.App -->
+    <PropertyGroup Condition="'$(_ReferencesMicrosoftAspNetCoreApp)' == ''">
+      <_ReferencesMicrosoftAspNetCoreApp>@(FrameworkReference->AnyHasMetadataValue('Identity', 'Microsoft.AspNetCore.App'))</_ReferencesMicrosoftAspNetCoreApp>
+    </PropertyGroup>
+  </Target>
+
+  <Target
+    Name="_DiscoverMvcApplicationParts"
+    Inputs="$(MSBuildAllProjects);$(ResolveAssemblyReferencesStateFile)"
+    Outputs="$(_MvcApplicationPartAttributeGeneratedFile)"
+    DependsOnTargets="$(_DiscoverMvcApplicationPartsDependsOn)"
+    Condition="'$(_ReferencesMicrosoftAspNetCoreApp)' == 'true'">
+
+    <ItemGroup>
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Abstractions" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.ApiExplorer" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Core" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Cors" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.DataAnnotations" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Formatters.Json" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Formatters.Xml" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Localization" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Razor" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.RazorPages" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.TagHelpers" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.ViewFeatures" />
+    </ItemGroup>
+
+    <DiscoverMvcApplicationParts ResolvedReferences="@(ReferencePath)" MvcAssemblyNames="@(_MvcAssemblyName)">
+      <Output TaskParameter="ApplicationPartAssemblyNames" ItemName="_ApplicationPartAssemblyNames"/>
+    </DiscoverMvcApplicationParts>
+
+    <ItemGroup>
+       <_MvcApplicationPartAttribute Include="Microsoft.AspNetCore.Mvc.ApplicationParts.ApplicationPart">
```

Comment:
Should this be `.ApplicationPartAttribute`? or doesn't matter?

Reply:
Yeah, we should do the needful.